lib/semantics/check-declarations.cpp

```diff
@@ -371,6 +375,14 @@ void CheckHelper::CheckObjectEntity(
       }
     }
   }
+  if (const DeclTypeSpec * type{details.type()}) {  // C708
+    if (type->IsPolymorphic() && !symbol.attrs().test(Attr::POINTER) &&
```

Comment:
Consider applying DeMorgan's law for a more intuitive expression: `type->IsPolymorphic() && !(x || y || z)`

Reply:
Thanks, Peter.  Will do.

Reply:
Using `IsAllocatableOrPointer` from tools.h would help to simplify it too.

Reply:
Thanks, Tim.  I'll use it.